### PR TITLE
[scudo][NFC] Explicit type casting to avoid compiler warning

### PR DESCRIPTION
**Pull request by ChiaHungDuan**

### CODE REVIEW

File: compiler-rt/lib/scudo/standalone/allocator_common.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ template <class SizeClassAllocator> struct TransferBatch {
   void moveNToArray(CompactPtrT *Array, u16 N) {
     DCHECK_LE(N, Count);
     memcpy(Array, Batch + Count - N, sizeof(Batch[0]) * N);
-    Count -= N;
+    Count -= static_cast<u16>(Count - N);
```
Comment (thurstond) [inline compiler-rt/lib/scudo/standalone/allocator_common.h:56]: These two lines are not equivalent. Should that be "Count = ..."?

Comment (ChiaHungDuan) [reply compiler-rt/lib/scudo/standalone/allocator_common.h:56]: oops, sorry my bad. I fixed that but didn't include it in the commit.

**Review by cferris1000 — APPROVED** [review]
LGTM

File: compiler-rt/lib/scudo/standalone/allocator_common.h (hunk 1 above)
Comment (ChiaHungDuan) [reply compiler-rt/lib/scudo/standalone/allocator_common.h]: Fixed. Thanks for catching it

**Review by thurstond — APPROVED** [review]

**Pull request merged**